app/src/main/java/com/eveningoutpost/dexdrip/webservices/WebServiceSgv.java

```diff
@@ -140,56 +187,69 @@ public WebResponse request(String query) {
                         item.put("rssi", 100);
                         item.put("type", "sgv");
                     }
-                    if (units_indicator > 0) {
-                        item.put("units_hint", Pref.getString("units", "mgdl").equals("mgdl") ? "mgdl" : "mmol");
-                        units_indicator = 0;
-                    }
 
-                    // emit the external status line once if present
-                    if (external_status_line.length() > 0) {
-                        item.put("aaps", external_status_line);
-                        item.put("aaps-ts", getLastStatusLineTime());
-                        external_status_line = "";
+                    if (!brief) {
+                        cachedJson.put(reading.uuid, item);
                     }
+                    withManual++;
+                    reply.put(item);
+                }
 
-                    // emit result code from steps if present
-                    if (steps_result_code > 0) {
-                        item.put("steps_result", steps_result_code);
-                        steps_result_code = 0;
-                    }
+                UserError.Log.d(TAG, "Processed "+withCache+" cached entries and "+withManual+" manual entries");
+                UserError.Log.d(TAG, "Output: " + reply.toString());
+            } catch (JSONException e) {
+                UserError.Log.wtf(TAG, "Got json exception: " + e);
+            }
+        }
 
-                    // emit result code from heart if present
-                    if (heart_result_code > 0) {
-                        item.put("heart_result", heart_result_code);
-                        heart_result_code = 0;
-                    }
+        // Add special fields for first SGV entry
+        if (reply.length() > 0) {
+            try {
+                // Copy the first item only to a new JSONObject as to not
+                // add these items to the cached json object
+                JSONObject origItem = reply.getJSONObject(0);
+                JSONObject item = new JSONObject(origItem.toString());
+                if (units_indicator > 0) {
+                    String unitsHint = Pref.getString("units", "mgdl").equals("mgdl") ? "mgdl" : "mmol";
```

Comment:
What should this check do? Should it ensure that either "mgdl" or "mmol" is returned? I thought only either of them is allowed in the preferences.

Reply:
Honestly looks like some kind of sanity check for if the preference didn't exist. I copied this from earlier in the method, doesn't seem incredibly worth changing since it doesn't hurt anything.

Reply:
The declaration is `String getString(final String pref, final String def)`. Only if the preference is not set, it returns "mgdl". If the preferences could not be initialized, an empty string is returned by `Pref.getString`. Hence, always "mmol" is returned with the exception of "mgdl".